### PR TITLE
Dijkstra algorithm implementation

### PR DESCRIPTION
**Pull request by kolosovpetro**
- [x] I have performed a self-review of my code
- [x] My code follows the style guidelines of this project
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Comments in areas I changed are up to date
- [x] I have added comments to hard-to-understand areas of my code
- [x] I have made corresponding changes to the README.md

### CODE REVIEW

**Comment by kolosovpetro** [comment]
This is a preview PR

File: Algorithms/Graph/Dijkstra/DijkstraAlgorithm.cs [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        private static void ValidateGraphAndStartVertex<T>(DirectedWeightedGraph<T> graph, Vertex<T> startVertex)
+        {
+            if (graph is null)
+            {
+                throw new InvalidOperationException($"Graph is null {nameof(graph)}.");
+            }
+
+            if (startVertex.Graph != null && !startVertex.Graph.Equals(graph))
+            {
+                throw new InvalidOperationException($"Vertex does not belong to graph {nameof(startVertex)}.");
```
Comment (siriak) [inline Algorithms/Graph/Dijkstra/DijkstraAlgorithm.cs:96]: Please use ArgumentException(nameof(startVertex))

**Review by siriak — CHANGES_REQUESTED** [review]
I have a couple of comments, everything else looks good, thanks!

**Review by siriak — APPROVED** [review]
LGTM, thanks!

**Pull request merged**